Sources/Actions/Admin/ACP.php

```diff
@@ -1526,66 +1526,82 @@ public static function getFileVersions(&$versionOptions)
 		$lang_dir = Theme::$current->settings['default_theme_dir'] . '/languages';
 
 		$version_info = [
+			'root_versions' => [],
 			'file_versions' => [],
 			'default_template_versions' => [],
 			'template_versions' => [],
 			'default_language_versions' => [],
 			'tasks_versions' => [],
 		];
 
-		// Find the version in SSI.php's file header.
-		if (!empty($versionOptions['include_ssi']) && file_exists(Config::$boarddir . '/SSI.php')) {
-			$fp = fopen(Config::$boarddir . '/SSI.php', 'rb');
-			$header = fread($fp, 4096);
-			fclose($fp);
-
-			// The comment looks rougly like... that.
-			if (preg_match('~\*\s@version\s+(.+)[\s]{2}~i', $header, $match) == 1) {
-				$version_info['file_versions']['SSI.php'] = $match[1];
-			}
-			// Not found!  This is bad.
-			else {
-				$version_info['file_versions']['SSI.php'] = '??';
-			}
-		}
-
-		// Do the paid subscriptions handler?
-		if (!empty($versionOptions['include_subscriptions']) && file_exists(Config::$boarddir . '/subscriptions.php')) {
-			$fp = fopen(Config::$boarddir . '/subscriptions.php', 'rb');
-			$header = fread($fp, 4096);
-			fclose($fp);
-
-			// Found it?
-			if (preg_match('~\*\s@version\s+(.+)[\s]{2}~i', $header, $match) == 1) {
-				$version_info['file_versions']['subscriptions.php'] = $match[1];
-			}
-			// If we haven't how do we all get paid?
-			else {
-				$version_info['file_versions']['subscriptions.php'] = '??';
-			}
-		}
+		$root_files = [
+			'cron.php',
+			'proxy.php',
+			'SSI.php',
+			'subscriptions.php'
+		];
 
-		// Load all the files in the Sources directory, except this file and the redirect.
-		$sources_dir = dir(Config::$sourcedir);
+		// Find the version in root files header.
+		if (!empty($versionOptions['include_root'])) {
+			foreach ($root_files as $file) {
+				if (!file_exists(Config::$boarddir . '/' . $file)){
+					continue;
+				}
 
-		while ($entry = $sources_dir->read()) {
-			if (substr($entry, -4) === '.php' && !is_dir(Config::$sourcedir . '/' . $entry) && $entry !== 'index.php') {
-				// Read the first 4k from the file.... enough for the header.
-				$fp = fopen(Config::$sourcedir . '/' . $entry, 'rb');
+				$fp = fopen(Config::$boarddir . '/' . $file, 'rb');
 				$header = fread($fp, 4096);
 				fclose($fp);
-
-				// Look for the version comment in the file header.
+	
```

Comment:
```suggestion

```

Reply:
My bad.  New editor, still getting used to it for things like copy and paste.